tools/rtpsrelay/RelayHandler.cpp

```diff
@@ -702,9 +723,75 @@ DataHandler::DataHandler(ACE_Reactor* reactor,
 : VerticalHandler(reactor, relay_addresses, association_table, responsible_relay_writer, responsible_relay_reader, lifespan, rtps_discovery, application_domain, application_participant_guid, crypto)
 {}
 
-std::string DataHandler::extract_relay_address(const RelayAddresses& relay_addresses) const
+ACE_INET_Addr DataHandler::extract_relay_address(const RelayAddresses& relay_addresses) const
+{
+  return ACE_INET_Addr(relay_addresses.data_relay_address().c_str());
+}
+
+StunHandler::StunHandler(ACE_Reactor* reactor)
+  : RelayHandler(reactor)
+{}
+
+void StunHandler::process_message(const ACE_INET_Addr& remote_address,
+                                  const OpenDDS::DCPS::MonotonicTimePoint&,
+                                  const OpenDDS::DCPS::Message_Block_Shared_Ptr& msg)
+{
+  OpenDDS::DCPS::Serializer serializer(msg.get(), true);
```

Comment:
It's odd to see a literal `true` here.  We want the serializer to swap bytes unconditionally on all architectures?

Reply:
STUN messages should always be in big-endian.  Let me know if my next commit achieves this.

Reply:
Maybe make a comment saying this then?

Reply:
Yes, it's good with `DCPS::Serializer::SWAP_BE`

Reply:
Are there similar cases in the Relay/STUN/ICE code?